force_torque_sensor_broadcaster/test/test_force_torque_sensor_broadcaster.cpp

```diff
@@ -40,15 +40,43 @@ constexpr auto NODE_ERROR = controller_interface::CallbackReturn::ERROR;
 
 void ForceTorqueSensorBroadcasterTest::SetUpTestCase() {}
 
-void ForceTorqueSensorBroadcasterTest::TearDownTestCase() {}
+void ForceTorqueSensorBroadcasterTest::TearDownTestCase()
+{
+  // Ensure all nodes are properly shutdown
+  rclcpp::shutdown();
+  // Add a small delay to allow proper cleanup
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
+}
 
 void ForceTorqueSensorBroadcasterTest::SetUp()
 {
   // initialize controller
   fts_broadcaster_ = std::make_unique<FriendForceTorqueSensorBroadcaster>();
 }
 
-void ForceTorqueSensorBroadcasterTest::TearDown() { fts_broadcaster_.reset(nullptr); }
+void ForceTorqueSensorBroadcasterTest::TearDown()
+{
+  // Reset the broadcaster with proper cleanup
+  if (fts_broadcaster_)
+  {
+    if (
+      fts_broadcaster_->get_lifecycle_state().id() !=
+      lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+    {
+      if (
+        fts_broadcaster_->get_lifecycle_state().id() ==
+        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE)
+      {
+        ASSERT_EQ(fts_broadcaster_->on_deactivate(rclcpp_lifecycle::State()), NODE_SUCCESS);
+      }
+      // Clean up the broadcaster
+      ASSERT_EQ(fts_broadcaster_->on_cleanup(rclcpp_lifecycle::State()), NODE_SUCCESS);
```

Comment:
Shouldn't we have called the lifecycle node's transition calls instead of the callbacks? The lifecycle node won't have the correct state afterwards.

Reply:
You are right!

Reply:
@Juliaj could you fix this in a new PR please?